Count of all fused MLA ops in conv microbenchmark

milpuz01 opened the pull request:
This patch updates how we count number of items that convolution microbenchmarks operates on. The number of items reported now it total number of fused multiply and accumulate that convolution executes.

Also added to use the wall clock to decide for how log to run the benchmark loop as we have noticed on higher number of cores total CPU consumption is not a good deciding factor and benchmark loop only runs couple of times or at worst only once.

milpuz01 committed:
Count of all fused MLA ops in conv microbenchmark

This patch updates how we count number of items that convolution microbenchmarks operates on. The number of items reported now it total number of fused multiply and accumulate that convolution executes.

Also added to use the wall clock to decide for how log to run the benchmark loop as we have noticed on higher number of cores total CPU consumption is not a good deciding factor and benchmark loop only runs couple of times or at worst only once.

penpornk reviewed (changes requested):
Thank you for the PR!

penpornk commented on tensorflow/core/kernels/conv_ops_benchmark_test.cc:
```diff
@@ -332,9 +332,10 @@ static Graph* FusedConv2DWithBatchNorm(
+// The number of items is equal to number of fused multiply and accumlate operations
+#define BM_SET_INFO(N, H, W, C, FW, FH, FC, type, LABEL, NAME)             \
   state.SetItemsProcessed(static_cast<int64_t>(state.iterations()) * (N) * \
-                          (H) * (W) * (C));                                \
+                          (H) * (W) * (FC) * (C) * (FW) * (FH));           \
```
Also, we usually count FMA as two operations.

gbaned commented:
@milpuz01 Can you please check @penpornk's comments and keep us posted ? Thank you!

cantonios commented on tensorflow/core/kernels/conv_ops_benchmark_test.cc:
```diff
@@ -346,11 +347,11 @@ static Graph* FusedConv2DWithBatchNorm(
   BENCHMARK(BM_NAME(BM_Conv2D, type, N, H, W, C, FW, FH, FC))           \
       ->Arg(/*unused arg*/ 1)                                           \
-      ->MeasureProcessCPUTime();
+      ->MeasureProcessCPUTime()->UseRealTime();
```
I don't love the idea of using different settings internally vs externally.

It sounds like the main issue is in the minimum benchmark time, but we can increase that via command-line parameter when running the benchmark.

milpuz01 replied:
Thanks @penpornk and @cantonios. As suggested I removed this because I can see now that I am able to increase iteration count by passing from command line for example `--benchmark_min_time`

gbaned commented:
Hi @milpuz01 Any update on this PR? Please. Thank you!

cantonios commented:
@milpuz01 is this still active?  If not, we should close it.

milpuz01 commented:
> @milpuz01 is this still active? If not, we should close it.

Hi @cantonios, yes this is still active. I will try to address @penpornk comments as soon as possible and update the PR. Sorry for delay in pushing this through.

Many thanks,
Milos.

gbaned commented:
Hi @milpuz01 Any update on this PR? Please. Thank you!

milpuz01 committed:
Address review comments

milpuz01 commented:
> Hi @milpuz01 Any update on this PR? Please. Thank you!

HI @gbaned, very sorry for very long delay in addressing review comments. I believe they are addressed with [069cb25](https://github.com/tensorflow/tensorflow/pull/58014/commits/069cb256f1555ec2b1db14da86ee2d8898c7a760)
